### ISSUE TITLE
Win64 LLVM 3.6.0 segfaults in umfzl_scale during linalg/arnoldi test

### ISSUE THREAD

**Issue by tkelman** [JuliaLang/julia#11085]
Seen in https://github.com/JuliaLang/julia/issues/10394#issuecomment-89682836 and a few other times in that thread, happens when running `make testall1` but not running the tests individually.
```
Tony@TK-samsung ~/julia
$ usr/bin/julia -e 'versioninfo()'
Julia Version 0.4.0-dev+4586
Commit 9aacf5e* (2015-05-01 14:40 UTC)
Platform Info:
  System: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-2630QM CPU @ 2.00GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.6.0

Tony@TK-samsung ~/julia
$ make testall1
    JULIA test/all
     * linalg1              in  90.89 seconds
     * linalg2              in  55.96 seconds
     * linalg3              in  24.75 seconds
     * linalg4              in   7.68 seconds
     * linalg/lapack        in   3.64 seconds
     * linalg/triangular    in 443.51 seconds
     * linalg/tridiag       in  13.26 seconds
     * linalg/bidiag        in  24.81 seconds
     * linalg/diagonal      in  13.51 seconds
     * linalg/pinv          in  15.80 seconds
     * linalg/givens        in   9.27 seconds
     * linalg/cholesky      in  32.98 seconds
     * linalg/lu            in  23.11 seconds
     * linalg/symmetric     in  21.49 seconds
     * linalg/arnoldi
Please submit a bug report with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ACCESS_VIOLATION at 0x2a32ed95f -- umfzl_scale at  (unknown line)
umfzl_scale at D:\cygwin64\home\Tony\julia\usr\bin\libumfpack.DLL (unknown line)
umfzl_scale_column at D:\cygwin64\home\Tony\julia\usr\bin\libumfpack.DLL (unknown line)
umfzl_kernel at D:\cygwin64\home\Tony\julia\usr\bin\libumfpack.DLL (unknown line)
umfpack_zl_numeric at D:\cygwin64\home\Tony\julia\usr\bin\libumfpack.DLL (unknown line)
umfpack_numeric! at .\sparse\umfpack.jl:179
lufact at .\sparse\umfpack.jl:112
factorize at .\sparse\linalg.jl:688
eigs at .\linalg\arnoldi.jl:88
unknown function (ip: -1547168413)
jl_apply_generic at /home/Tony/julia/src\gf.c:1721
unknown function (ip: -1548220569)
unknown function (ip: -1548220218)
eigs at .\linalg\arnoldi.jl:5
unknown function (ip: -1546583741)
anonymous at .\no file:42
jl_apply at /home/Tony/julia/src\julia.h:1276
jl_toplevel_eval_flex at /home/Tony/julia/src\toplevel.c:549
jl_load_ at /home/Tony/julia/src\toplevel.c:601
runtests at D:\cygwin64\home\Tony\julia\test\testdefs.jl:78
unknown function (ip: 523502759)
jl_apply_generic at /home/Tony/julia/src\gf.c:1723
jl_f_apply at /home/Tony/julia/src\builtins.c:470
anonymous at .\multi.jl:626
run_work_thunk at .\multi.jl:587
remotecall_fetch at .\multi.jl:660
jl_apply_generic at /home/Tony/julia/src\gf.c:1723
jl_f_apply at /home/Tony/julia/src\builtins.c:470
remotecall_fetch at .\multi.jl:675
jl_apply_generic at /home/Tony/julia/src\gf.c:1723
jl_f_apply at /home/Tony/julia/src\builtins.c:470
anonymous at .\task.jl:1390
start_task at /home/Tony/julia/src\task.c:232
Makefile:9: recipe for target 'all' failed
make[1]: *** [all] Error 1
Makefile:492: recipe for target 'testall1' failed
make: *** [testall1] Error 2
```

Possibly a separate issue, but trying to run 3 test workers in parallel for the same build threw a really strange exception in the dates test: https://gist.github.com/tkelman/201a285459393fa39e9a

**Comment by tkelman** [comment]
Also happens on 3.6.1.

**Comment by tkelman** [comment]
Seems to have stopped happening. I'd do a reverse bisect and figure out what fixed it, if that didn't take so long.

**Comment by tkelman** [comment]
good news is all tests up to replcompletions (#11992) pass, but bad news is the performance here https://gist.github.com/tkelman/12b11a124f3b343d70bb
